robot/web/web_c.go

```diff
@@ -84,150 +82,83 @@ func (svc *webService) streamInitialized() bool {
 
 func (svc *webService) addNewStreams(ctx context.Context) error {
 	if !svc.streamInitialized() {
-		return nil
+		return errors.New("stream server not initialized")
 	}
+
+	// Refreshing sources will walk the robot resources for anything implementing the camera and
+	// audioinput APIs and mutate the `svc.videoSources` and `svc.audioSources` maps.
 	svc.refreshVideoSources()
 	svc.refreshAudioSources()
+
 	if svc.opts.streamConfig == nil {
+		// The `streamConfig` dictates the video and audio encoder libraries to use. We can't do
+		// much if none are present.
 		if len(svc.videoSources) != 0 || len(svc.audioSources) != 0 {
-			svc.logger.Debug("not starting streams due to no stream config being set")
+			svc.logger.Warn("not starting streams due to no stream config being set")
```

Comment:
I bumped this to `Warn`. I don't expect this to be noisy. And if we were hitting this scenario, I expect a user would like to know why they can't stream stuff without having to turn on (global) debug logging.

Reply:
Good call. We should not get into this scenario, but if so, we have problems.